### PR TITLE
Add timestamp to job temp folder and job ID

### PR DESCRIPTION
**Pull request by homezcx**

### CODE REVIEW

**Review by ryumbra — APPROVED** [review]

**Comment by ryumbra** [comment]
On a second thought, let's make sure that Azure Batch Job ID that we use is guaranteed to be unique. I know that's the case for HelloReef.cs, because we specifically set application id to include the date, but if the client just uses something like "tlcpp", then we'll have conflicts in Azure Batch.

**Comment by homezcx** [comment]
Yes. Azure Batch requires unique Job ID as well. Patched in 2nd commit.

**Review by ryumbra — APPROVED** [review]

File: lang/cs/Org.Apache.REEF.Client/AzureBatch/AzureBatchDotNetClient.cs [hunk 1]
```diff
@@ -78,7 +78,8 @@ public Task<FinalState> GetJobFinalStatus(string appId)
         public void Submit(JobRequest jobRequest)
         {
             var configModule = AzureBatchClientConfiguration.ConfigurationModule;
-            string jobId = jobRequest.JobIdentifier;
+            var timestamp = DateTime.Now.ToString("yyyyMMddHHmmssfff");
+            string jobId = string.Join("-", jobRequest.JobIdentifier, timestamp);
```
Comment (ryumbra) [inline lang/cs/Org.Apache.REEF.Client/AzureBatch/AzureBatchDotNetClient.cs]: One thing to keep in mind is that there is a limit to how long Azure Batch job ID can be. So if jobRequest.JobIdentifier is too long, we might exceed it.

Comment (homezcx) [reply lang/cs/Org.Apache.REEF.Client/AzureBatch/AzureBatchDotNetClient.cs]: Updated as what Java did. For local tmp folder, it is still doing with timestamp for user to check.

Comment (homezcx) [reply lang/cs/Org.Apache.REEF.Client/AzureBatch/AzureBatchDotNetClient.cs]: fixed.

**Pull request merged**